client/ArtifactsUIController.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * ArtifactsUIController.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+ #pragma once
+
+#include "../lib/constants/EntityIdentifiers.h"
+#include "../lib/networkPacks/ArtifactLocation.h"
+
+VCMI_LIB_NAMESPACE_BEGIN
+
+class CGHeroInstance;
+
+VCMI_LIB_NAMESPACE_END
+
+class ArtifactsUIController
+{
+public:
+	size_t numOfMovedArts;
+	size_t numOfArtsAskAssembleSession;
+	std::set<ArtifactID> ignoredArtifacts;
+
+	boost::mutex askAssembleArtifactMutex;
```

Comment:
Do we need mutex? All UI-related events are already protected by UI mutex

Reply:
Using a mutex was the simplest solution (maybe not the best). Because there is no need to implement custom waiting procedures.